### DISCUSSION TITLE
CharacterPromotions description

### DISCUSSION THREAD

**Discussion by latot** [Mar-7th/StarRailRes#21] [General]
Hi!, actually in no page of the docs, the character promotions (which I assume are for base stats).

So, the basic question is, for a character, at a particular ascension and a given level, how to calculate the actual stat for it?

I imagine, would be like ```char.value[ascension_lvl].base + char.value[ascension_lvl].step*level```, is this correct?

**Comment by NoahFsc** [comment]
Hello, have you found a solution ? Doing base stat + step * levels doesn't really work because there are not the ascension values added to it. However, I don't know where to get these.

**Reply by nathacks** [reply]
Hello, several months ago, I made a website showing the different statistics of a character, based on the image I sent you. Is that what you're interested in?

<img width="297" alt="Capture d’écran 2025-06-26 à 15 59 47" src="https://github.com/user-attachments/assets/8924aab1-63ad-4e20-8dc0-3f1933f4f837" />

If so, I will send you the calculations I use.

The URL of the site, in case you’re interested in something else: https://hsr.naths.dev/#/menu/characters

**Reply by NoahFsc** [reply]
Thanks ! I think I found out how the stats are calculated today.

**Comment by sadfaceman-gt** [comment]
These two idiots didn't even give the answer. Base stat is calculated using :

Base of current rank + Step * (Level - 1)

For example level 36 Acheron base hp is 275.616 + (7.676 * 35) = 544.276, stats are displayed round up so it becomes 545
